application-workloads/darktrace/darktrace-vsensor-autoscaling/azuredeploy.json

```diff
@@ -0,0 +1,1141 @@
+                                }
+                            },
+                            {
+                                "name": "healthRepairExtension",
+                                "properties": {
+                                    "autoUpgradeMinorVersion": true,
+                                    "publisher": "Microsoft.ManagedServices",
+                                    "type": "ApplicationHealthLinux",
+                                    "typeHandlerVersion": "1.0",
+                                    "settings": {
+                                        "protocol": "https",
+                                        "port": 443,
+                                        "requestPath": "/healthcheck"
+                                    }
+                                }
+                            },
+                            {
+                                "name": "DarktraceVSensorInstall",
+                                "properties": {
+                                  "publisher": "Microsoft.Azure.Extensions",
+                                  "type": "CustomScript",
+                                  "typeHandlerVersion": "2.1",
+                                  "autoUpgradeMinorVersion": true,
+                                  "settings": {
+                                    "skipDos2Unix":false,
+                                    "timestamp":123456789          
+                                  },
+                                  "protectedSettings": {
+                                     "script": "[base64(concat('#! /bin/bash -e\n\nwait_for_apt() {\n    echo \"Waiting for apt/dpkg lock.\"\n    tries=0\n    maxtries=10\n    while sudo fuser /var/lib/dpkg/lock /var/lib/apt/lists/lock /var/cache/apt/archives/lock >/dev/null 2>&1 && [ $tries -le $maxtries ]; do\n       sleep 20\n       ((tries=tries+1))\n    done\n    if [ $tries -ge $maxtries ]; then\n        echo \"ERROR: Failed to get apt / dpkg lock before timeout. Please wait and try again, or fix dpkg lock manually.\"\n        exit 1\n    fi\n}\nexec > >(tee -a /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1\nif [ \"', parameters('VSensorUpdateKey'), '\" = \"updateKey-example\" ]; then\n    echo \"Skipping install in azure CI due to dummy key.\"\n    exit 0\nfi\nwait_for_apt\nsleep 10\nwait_for_apt\nbash <(wget https://packages.darktrace.com/install -O -) --updateKey \"', parameters('VSensorUpdateKey'), '\"\n/usr/sbin/set_pushtoken.sh \"', parameters('appliancePushtoken'), '\" \"', parameters('applianceHostName'), ':', parameters('appliancePort'), '\"\nset_ossensor_hmac.sh \"', parameters('OSSensorHMACToken'), '\"\nset_pcap_azure_container.sh \"', variables('blobStorageAccountName'), '\" \"', variables('blobStorageContainerName'), '\"\nset_ossensor_loadbalancer_direct.sh 1\nset_ephemeral.sh 1\necho \"Configuration complete, vSensor is ready for use.\"\n'))]"
```

Comment:
are these bash scripts getting pulled from `packages.darktrace.com`?

Reply:
Yes! Thanks for reviewing.